### DISCUSSION TITLE
Code for I local deployment suggestion

### DISCUSSION THREAD

**Discussion by pabloto** [codefori/vscode-ibmi#1234] [General]
Hi, 

I'm starting in deploying in local with my team, we have a local gitlab server, now we have still some problem with deployment, for the moment the idea is than the main branch of our repository is the branch relative to what we have in production, so for a new project or bug fix we create a new branch and in our test server we start to deploy and compile.

Now we are encountering some configuration problem, because the deploy directory is the same for main branch and fix branch or some project relative branch.
Is there the possibility to set some env variable at branch level to set this kind of config:

1. Deploy directory to IFS
2. Compile library (so the production library remain the as in production server, and the library relative to fix bug or project is different)

So switching between branch I didn't have to remember to change the deploy directory and the current lib.

Many thanks

**Comment by worksofliam** [comment]
@pabloto excellent idea. I was just thinking about how we could have branch specific variables for things like library name (that would be the branch library) etc.

**Reply by sebjulliand** [reply]
Maybe we could support an .env file at the root of the project that would be merged at runtime with the one at the workspace level. The branch .env would override values from the workspace's.

**Comment by andreas-prouza** [comment]
I have set up a github project for this: https://github.com/andreas-prouza/ibm-i-build
There I use GNU Make for building the objects.
With that you can have a common build rule and a ".makeprofile.mk" file which is in the git ignore list.
So every developer can choose his own ifs directory and target lib directory for compiling.
For compile for test or production system, I have a separate directory in the ifs (e.g. /apps/erp/test and /apps/erp/prod) where the definition of each environment is defined in the ".makeprofile.mk".

So when I am developing some code, I compile based on my own home directory in my own developer lib.
When I am ready for testing, I merge my branch/changes to the test branch, pull the changes the ifs directory (e.g. /apps/erp/test) and trigger building there.
The same for production.
To be comfortable I set up some vscode commands I need for my projects:

![image](https://user-images.githubusercontent.com/65070910/231663394-316e8ebf-79f8-424c-b455-080d6341b73a.png)

**Reply by worksofliam** [reply]
@andreas-prouza are you interested in having your page updated? Code for IBM i does a bunch of stuff regarding uploading files, .env support (whereas you have env.mk, great work there). 

Of course it's specific to Code for IBM i, but we're trying to simplify local development:

https://halcyon-tech.github.io/docs/#/pages/developing/local/getting-started

**Reply by andreas-prouza** [reply]
Sure, I am always open to suggestions.
But what exactly do you want me to update?

**Reply by worksofliam** [reply]
@andreas-prouza PR coming this weekend from me! (If I find the time!)

**Comment by tehTK** [comment]
Hi @worksofliam , 
any update or plans for variables that can change the 1. Deploy Directory and 2. Build Library ? 

It would be fantastic if the following functionality were possible: When switching to a specific Git branch, the build library variable could automatically update to match the branch name. For instance, when I switch to the 'DVL0001' branch, the build library would also change to the 'DVL0001' library (or any custom name that I can configure based on my branch naming). This would enable me to deploy my changes and compile my objects without manual intervention. Currently, I have to manually remember to update my CURLIB to the branch-specific library, which can be error-prone
